src/core.jl

```diff
@@ -87,101 +235,161 @@ function issubset{T<:SimpleGraph}(g::T, h::T)
 end
 
 """Add a new vertex to the graph `g`."""
-function add_vertex!(g::SimpleGraph)
-    n = length(vertices(g)) + 1
-    g.vertices = 1:n
-    push!(g.badjlist, Int[])
-    push!(g.fadjlist, Int[])
+function add_vertex!(g::SparseGraph)
+    g.fm.m += 1
+    g.fm.n += 1
+    push!(g.fm.colptr, g.fm.colptr[end])
+    return g.fm.m
+end
 
-    return n
+function add_vertex!(g::SparseDiGraph)
+    g.fm.m += 1
+    g.fm.n += 1
+    push!(g.fm.colptr, g.fm.colptr[end])
+    g.bm.m += 1
+    g.bm.n += 1
+    push!(g.bm.colptr, g.bm.colptr[end])
+    return g.fm.m
 end
 
+
 """Add `n` new vertices to the graph `g`."""
-function add_vertices!(g::SimpleGraph, n::Integer)
-    for i = 1:n
-        add_vertex!(g)
-    end
-    return nv(g)
+function add_vertices!(g::SparseGraph, n::Integer)
+    g.fm.m += n
+    g.fm.n += n
+    append!(g.fm.colptr, fill(g.fm.colptr[end], n))
+    return g.fm.m
+end
+
+function add_vertices!(g::SparseDiGraph, n::Integer)
+    g.fm.m += n
+    g.fm.n += n
+    append!(g.fm.colptr, fill(g.fm.colptr[end], n))
+    g.bm.m += n
+    g.bm.n += n
+    append!(g.bm.colptr, fill(g.bm.colptr[end], n))
+    return g.fm.m
 end
 
 """Return true if the graph `g` has an edge from `src` to `dst`."""
-has_edge(g::SimpleGraph, src::Int, dst::Int) = has_edge(g,Edge(src,dst))
+has_edge(g::SimpleGraph, e::Edge) = has_edge(g, src(e), dst(e))
+has_edge(g::SimpleGraph, s::Int, d::Int) = d <= nv(g) && s <= nv(g) && fmat(g)[d, s]
```

Comment:
This suggests that if the element `[d,s]` is true, that means that the rows are the destinations of each edge, and the columns are sources, is that correct?